File: kafka-connect-common/src/main/scala/io/lenses/streamreactor/common/config/base/traits/BaseSettings.scala
/*
 * Copyright 2017-2024 Lenses.io Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.lenses.streamreactor.common.config.base.traits

import java.util

import org.apache.kafka.common.config.types.Password

trait WithConnectorPrefix {
  def connectorPrefix: String

}
trait BaseSettings extends WithConnectorPrefix {

  def getString(key: String): String

  def getInt(key: String): Integer

  def getBoolean(key: String): java.lang.Boolean

  def getPassword(key: String): Password

  def getList(key: String): util.List[String]
}

File: java-connectors/kafka-connect-common/src/main/java/io/lenses/java/streamreactor/common/config/base/intf/BaseSettings.java
package io.lenses.java.streamreactor.common.config.base.intf;

import java.util.List;
import org.apache.kafka.common.config.types.Password;

/**
 * Interface that exposes methods to fetch Properties of different types.
 */
public interface BaseSettings extends ConnectorPrefixed {
  String getString(String key);

  Integer getInt(String key);

  Boolean getBoolean(String key);

  Password getPassword(String key);

  List<String> getList(String key);


}
